### PR TITLE
feat(icon/pill): add custom color support - FE-3056

### PR DESCRIPTION
**Pull request by mkrds**
### Proposed behaviour
Icon - add possibility to change Icon `color` through the `color` prop and `background-color` through the `bg` prop
Pill - add possibility to change Pill `border-color` and `background-color` (when `fill` is true) through the `borderColor` prop.

### Checklist

- [x] Commits follow our style guide
- [x] Screenshots are included in the PR if useful
- [x] All themes are supported if required
- [x] Unit tests added or updated if required
- [x] Cypress automation tests added or updated if required
- [x] Storybook added or updated if required
- [x] Typescript `d.ts` file added or updated if required
- [x] Carbon implementation and Design System documentation are congruent

### Testing instructions
Pill and Icon examples in `mdx` files or knobs in Test stories

### CODE REVIEW

File: src/components/icon/icon.stories.mdx
Comment (sjohnson-sage) [inline src/components/icon/icon.stories.mdx:129]: I don't think this is being used in this story

File: src/components/pill/pill.style.js
Comment (sjohnson-sage) [inline src/components/pill/pill.style.js:39]: Should this feed an error to the user to tell them why their colour was changed to white?

**Review by sjohnson-sage — COMMENTED** [review]
Looks good, just a couple of questions before I approve

File: src/components/pill/pill.style.js
Comment (sjohnson-sage) [reply src/components/pill/pill.style.js:39]: Do we even need this in the code? Should really be down to the designer or dev using the component to make sure their colours are accessible

Comment (mkrds) [reply src/components/pill/pill.style.js:39]: `contentColor` = color of the text in the `Pill`
It has to be automated because user can control the color of the Pill but not the color of the pill text

**Review by nicktitchmarsh — APPROVED** [review]

File: src/components/icon/icon.style.js
Comment (dawidzarzycki) [reply src/components/icon/icon.style.js:128]: Just a suggestion, I think it will be more readable but I leave it to your decision

**Review by sjohnson-sage — APPROVED** [review]

File: src/components/pill/pill.component.js
Comment (dawidzarzycki) [inline src/components/pill/pill.component.js:65]: I think we can remove the spaces between propTypes

**Review by dawidzarzycki — APPROVED** [review]

**Review by sjohnson-sage — APPROVED** [review]

**Review by nicktitchmarsh — APPROVED** [review]

**Review by dawidzarzycki — APPROVED** [review]

**Review by dawidzarzycki — APPROVED** [review]

**Review by nicktitchmarsh — DISMISSED** [review]

File: src/components/icon/icon.spec.js
Comment (sjohnson-sage) [inline src/components/icon/icon.spec.js:153]: Could you put `// eslint-disable-next-line no-console` above these `console.error` please? To stop the lint warnings

File: src/components/icon/icon.spec.js
Comment (sjohnson-sage) [inline src/components/icon/icon.spec.js:160]: Could you put `// eslint-disable-next-line no-console` above these `console.error` please? To stop the lint warnings

File: src/components/icon/icon.stories.mdx
Comment (sjohnson-sage) [inline src/components/icon/icon.stories.mdx:126]: Is this story supposed to be disabled in chromatic? Might be useful to have snapshots of these props working

File: src/components/pill/pill.spec.js
Comment (sjohnson-sage) [inline src/components/pill/pill.spec.js:144]: Same comment as above for the lint warnings

File: src/components/icon/icon.stories.mdx
Comment (DlgSHi) [reply src/components/icon/icon.stories.mdx:126]: I think we need to have it in `All` story as `visual` part

Comment (mkrds) [reply src/components/icon/icon.stories.mdx:126]: @sjohnson-sage I've added these to `All` story

Comment (sjohnson-sage) [reply src/components/icon/icon.stories.mdx:126]: Thanks

**Review by sjohnson-sage — APPROVED** [review]

**Review by DlgSHi — APPROVED** [review]

**Review by nicktitchmarsh — APPROVED** [review]

**Review by dawidzarzycki — APPROVED** [review]

**Pull request merged**
